src/mean.ts

```diff
@@ -0,0 +1,87 @@
+import type { IterableContainer } from "./internal/types";
+import { purry } from "./purry";
+import { sum } from "./sum";
+
+type Mean<T extends IterableContainer<bigint> | IterableContainer<number>> =
+  // Empty arrays would always result in undefined.
+  T extends readonly []
+    ? undefined
+    : // Non-empty bigint arrays will always result in bigint.
+      T extends readonly [bigint, ...ReadonlyArray<unknown>]
+      ? bigint
+      : // But an empty bigint array would result in undefined.
+        T[number] extends bigint
+        ? undefined
+        : // Non-bigint arrays are always handled correctly.
+          number;
+
+/**
+ * Returns the mean of the elements of an array.
+ *
+ * Works for both `number` and `bigint` arrays, but not arrays that contain both
+ * types.
+ *
+ * IMPORTANT: The result for empty arrays would be `undefined`, regardless of
+ * the type of the array. This approach improves type-checking and ensures that
+ * cases where `NaN` might occur are handled properly. To avoid adding this to
+ * the return type for cases where the array is known to be non-empty you can use
+ * `hasAtLeast` or `isEmpty` to guard against this case.
```

Comment:
**question:** what do you think about adding a type test for something like

```ts
const data = [] as Array<number>;
if (hasAtLeast(data, 1)) {
  expectTypeOf(mean(data)) // etc
}
```

Reply:
Nice idea, that would be very useful!